### DISCUSSION TITLE
Testing changes to the debug server java extension

### DISCUSSION THREAD

**Discussion by sbj42** [microsoft/vscode-java-debug#1138] [Q&A] [answered]
I'm trying to test a change to the `java-debug` project.  Following the instructions in `CONTRIBUTING.md`, I:
* Run `npm install` and `npm run build-server` in `vscode-java-debug`
* Open `vscode-java-debug` in vscode and press F5 to start debugging the extension (after fixing the watch task, see #1136)

This all seems to work, so next I:
* Close the extension host
* Make a change in `java-debug`
* Run `npm run build-server` again
* Re-launch the extension host

But the change is not present.  The reason is that the `java-debug` OSGI bundle has the same `Bundle-Version` as before, and the Eclipse JDT Language Server won't update the installed bundle unless the version changes. See [BundleUtils.java#L125](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java#L125).

It seems the only way to test my change is to search in `java-debug` for the version string and replace it with a new one (e.g. `0.35.0` -> `0.35.1`).  It seems annoying to have to do that each time I try out a change to the java extension.

Is there a trick to it that I'm missing?  Or is this just something you live with while working on the extension?

**Comment by testforstephen** [comment, marked as answer]
In theory, it's better to add a qualifier to java-debug bundle version, but that would update vscode-java-debug package.json again and again. In current mechanism, there is a workaround to clean the OSGI bundle cache to allow new changes to take effect.

Here is the cache directories, you can delete those highlighting folders except for config.ini.

`C:\Users\<username>\.vscode\extensions\redhat.java-1.4.0\server\config_win`

![image](https://user-images.githubusercontent.com/14052197/158499982-836d3887-9d30-4b7b-9d56-99ed450a9c6e.png)

**Reply by sbj42** [reply]
Good enough for me, thanks!
